src/main/java/com/google/sps/servlets/AuthorizationServlet.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* This class handles authorization of the user*/
+
+@WebServlet("/authorization")
+public class AuthorizationServlet extends HttpServlet {
+
+  public enum Type {
+    RENTER,
+    HOST,
+    BOTH,
+    UNKNOWN;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    PrintWriter out = response.getWriter();
+
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Creating a Query to search through the datastore
+    Query query = new Query("User Data");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    final String emailLogin = request.getParameter("email");
+    final String pswLogin = request.getParameter("pswd");
```

Comment:
Are passwords required for users with a Google account? I think this part can be purely handled by authservice. Otherwise we would have to look into salting + hashing these passwords as it's not a good idea to store them plaintext

Reply:
Yes, I had a password section for the demonstration for our prototype but with me now implementing the Google Auth API I am not including the password section anymore. Thanks!

Reply:
Ok, awesome!